command-line-interface.md

```diff
@@ -0,0 +1,150 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
+That executable MUST support commands with the following template:
+
+```sh
+$ funC [global-options] <COMMAND> [command-specific-options] <command-specific-arguments>
+```
+
+## Global options
+
+None are required, but the runtime MAY support options that start with at least one hyphen.
+Global options MAY take positional arguments (e.g. `--log-level debug`).
+Command names MUST NOT start with hyphens.
+The option parsing MUST be such that `funC <COMMAND>` is unambiguously an invocation of `<COMMAND>` (even for commands not specified in this document).
+If the runtime is invoked with an unrecognized command, it MUST exit with a nonzero exit code and MAY log a warning to stderr.
+
+## Character encodings
+
+This API specification does not cover character encodings, but runtimes SHOULD conform to their native operating system.
+For example, POSIX systems define [`LANG` and related environment variables][posix-lang] for [declaring][posix-locale-encoding] [locale-specific character encodings][posix-encoding], so a runtime in an `en_US.UTF-8` locale SHOULD write its [version](#version) to stdout in [UTF-8][].
+
+## Commands
+
+### version
```

Comment:
On Sun, Jul 10, 2016 at 08:54:30AM -0700, Mike Brown wrote:
> > +### version
>
> While I completely agree with your preference, I don't think we
> should pick one that makes runc v1 non-compliant. So that leaves us
> with convincing the runc guys to accept a PR to add a version
> command (easy to do, I'll do it, nbd... but the runc 1.0 rc is
> already out) or go with 2 or 3.

A third option is to supply a third-party oci-cli-runc wrapper that
translates the ‘version’ command into a ‘runC --version’ call and
otherwise just passes arguments through to runC directly.  My
preferences here are:

1. runC accepts a patch to support both --version and ‘version’.
2. The OCI provides an oci-cli-runc wrapper.  This can fall under the
   compliance-test project or the runC project or a new OCI Project,
   it doesn't matter to me.  It's probably going to be about five
   lines of shell, if this is the only wrapping we have to do ;).
3. Go with spec approach 2 (require only --version).
4. Go with spec approach 3 (require either --version or ‘version’).